src/solver/impls/petsc/petsc.cxx

```diff
@@ -514,9 +514,7 @@ int PetscSolver::init() {
   CHKERRQ(ierr);
   ierr = ISColoringDestroy(&iscoloring);
   CHKERRQ(ierr);
-  ierr = MatFDColoringSetFunction(matfdcoloring,
-                                  reinterpret_cast<PetscErrorCode (*)()>(solver_f), this);
-  CHKERRQ(ierr);
+  throw BoutException("Coloring is not working");
```

Comment:
What's the issue here? If we know this doesn't work, can we remove it entirely? Or at least error when we set the option?